atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/render/TableClassRendererV2.java

```diff
@@ -0,0 +1,477 @@
+                ClassName.get(getColumnTypeClass(col))));
+
+        getterBuilder
+                .addStatement("$T colSelection = \n "
+                                + "$T.create($T.singletonList($T.toCachedBytes($S)))",
+                        ColumnSelection.class, ColumnSelection.class, Collections.class,
+                        PtBytes.class, col.getShortName())
+                .addStatement("$T<$T> rows = $T\n"
+                                + ".newArrayList(rowKeys)\n"
+                                + ".stream()\n"
+                                + ".map($T::of)\n"
+                                + ".collect($T.toList())",
+                        List.class, rowType, Lists.class, rowType, Collectors.class)
+                .addStatement("$T<byte[], $T<byte[]>> results = "
+                                + "t.getRows(tableRef, $T.persistAll(rows), colSelection)",
+                        SortedMap.class, RowResult.class, Persistables.class)
+                .addStatement("return results\n"
+                                + ".values()\n"
+                                + ".stream()\n"
+                                + ".map(entry -> $T.of(entry))\n"
+                                + ".collect($T.toMap(\n"
+                                + "     entry -> entry.getRowName().get$L(), \n"
+                                + "     $T::get$L))",
+                        rowResultType, Collectors.class, CamelCase(rowComponent.getComponentName()),
+                        rowResultType, VarName(col));
+
+        return getterBuilder.build();
+    }
+
+    private MethodSpec renderNamedGetSeveralRowsFromRowObjects(NamedColumnDescription col) {
+        MethodSpec.Builder getterBuilder = MethodSpec.methodBuilder("get" + VarName(col) + 's')
+                .addModifiers(Modifier.PUBLIC)
+                .addParameter(ParameterizedTypeName.get(ClassName.get(Iterable.class), rowType), "rowKeys");
+
+        getterBuilder.returns(ParameterizedTypeName.get(
+                ClassName.get(Map.class),
+                rowType,
+                ClassName.get(getColumnTypeClass(col))));
+
+        getterBuilder
+                .addStatement("$T colSelection = \n "
+                                + "$T.create($T.singletonList($T.toCachedBytes($S)))",
+                        ColumnSelection.class, ColumnSelection.class, Collections.class,
+                        PtBytes.class, col.getShortName())
+                .addStatement("$T<byte[], $T<byte[]>> results = "
+                                + "t.getRows(tableRef, $T.persistAll(rowKeys), colSelection)",
+                        SortedMap.class, RowResult.class, Persistables.class)
+                .addStatement("return results\n"
+                                + ".values()\n"
+                                + ".stream()\n"
+                                + ".map(entry -> $T.of(entry))\n"
+                                + ".collect($T.toMap(\n"
```

Comment:
do these newlines actually result in the correct indentation in the output?

Reply:
Yep, I added a sample generated API for the CAS table where the newlines display as expected.